### DISCUSSION TITLE
FLUX clip_l, t5xxl_fp16.safetensors, t5xxl_fp8_e4m3fn.safetensors

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [comfyanonymous/ComfyUI#4222] [General]
In anticipation of a flood of questions I am putting this here, to use FLUX you need the models in this link:
https://huggingface.co/comfyanonymous/flux_text_encoders/tree/main

Currently it is totaly incomprehensible which model is the CLIP_l in the model browser (VIT_L maybe?) and whether the two google ones are in the model browser the correct one is a guess too only the larger google model is inconsistent with the size of the one on hugging face the other seems to correlate and therefore confirm it likely for both of them.

![image](https://github.com/user-attachments/assets/37805bba-9c02-4982-8e05-3d0e03585b9a)

So for anyone that is about to get here because they downloaded a workflow that was made using the Hugging Face names, now you know, updates on the CLIP_l will follow below.

(currently all workflow naming I have found on CIVITAI are inconsistent with the naming in the model browser)

**Comment by ltdrdata** [comment]
Yup. you can use google-t5 model for FLUX.

**Reply by dhhunter** [reply]
I was looking through the Flux.1 Source on GitHub and in /tree/main/src/flux/util.py, I found the following:

`
def load_t5(device: str | torch.device = "cuda", max_length: int = 512) -> HFEmbedder:
    # max length 64, 128, 256 and 512 should work (if your sequence is short enough)
    return HFEmbedder("google/t5-v1_1-xxl", max_length=max_length, torch_dtype=torch.bfloat16).to(device)

def load_clip(device: str | torch.device = "cuda") -> HFEmbedder:
    return HFEmbedder("openai/clip-vit-large-patch14", max_length=77, torch_dtype=torch.bfloat16).to(device)
`

So I would assume they are using openai/clip-vit-large-patch14 and google/t5-v1_1-xxl!
